tests/checks/compare_locales.py

```diff
@@ -0,0 +1,364 @@
+    """
+    with pytest.raises(UnsupportedResourceTypeError):
+        cast_to_compare_locales('.random_ext', None, None, None)
+
+
+@pytest.mark.django_db
+def test_cast_to_properties_singular(entity_with_comment, translation0):
+    assert cast_to_compare_locales('.properties', entity_with_comment, translation0.string, 0) == (
+        ComparePropertiesEntity(
+            'KeyEntity0',
+            'entity0',
+            'entity0',
+            CommentEntity('example comment'),
+        ),
+        ComparePropertiesEntity(
+            'KeyEntity0',
+            'Translation for entity0',
+            'Translation for entity0',
+            CommentEntity('example comment'),
+        )
+    )
+
+
+@pytest.mark.django_db
+def test_cast_to_properties_plural(plural_entity, plural_translation):
+    assert cast_to_compare_locales('.properties', plural_entity, plural_translation.string, 1) == (
+        ComparePropertiesEntity(
+            'KeyEntity0',
+            'plural entity0',
+            'plural entity0',
+            CommentEntity('example comment'),
+        ),
+        ComparePropertiesEntity(
+            'KeyEntity0',
+            'Plural translation for entity0',
+            'Plural translation for entity0',
+            CommentEntity('example comment'),
+        )
+    )
+
+
+@pytest.mark.parametrize(
+    'quality_check_args', (
+        mock_quality_check_args(
+            resource_ext='properties',
+            string='Foobar2',
+            translation='Barfoo2'
+        ),
+        mock_quality_check_args(
+            resource_ext='properties',
+            string='Mozilla',
+            string_plural='Mozillas',
+            translation='Allizom',
+            plural_form=1
+        ),
+    )
+)
+def test_valid_translations(quality_check_args):
+    """
+    Quality check should return an empty dict if there's no errors.
+    """
+    assert quality_check(**quality_check_args) == {}
+
+
+@pytest.mark.parametrize(
+    'quality_check_args,failed_checks',
+    (
+        (
+            mock_quality_check_args(
+                resource_ext='properties',
+                string='%s Foo %s bar %s',
+                translation='%d Bar %d foo %d \q %',
+            ),
+            {
+                'clWarnings': [u'unknown escape sequence, \\q'],
+                'clErrors': ['Found single %']
+            }
+        ),
+
+        # Plural form should be matched against the string plural
+        (
+            mock_quality_check_args(
+                resource_ext='properties',
+                string='%s Foo %s bar %s %',
+                string_plural='Plural setting  %s \p',
+                translation='%d Bar %d foo %d',
+                plural_form=1
+            ),
+            {
+                'clErrors': [u'argument 1 `d` should be `s`']
+            }
+        ),
+        (
+            mock_quality_check_args(
+                resource_ext='properties',
+                string='Invalid #1 entity',
```

Comment:
I think it would be good to test for a real plural form, e.g. `Invalid #1 entity;Invalid #1 entities` and a translation with `Invalid translation;Invalid translations`.

Also, is it possible to test and trigger the warning for [incorrect number of plurals](https://hg.mozilla.org/l10n/compare-locales/file/tip/compare_locales/checks.py#l108)? You would need to pass the locale, e.g. the string above translated in french as "Invalid #1 entities" should trigger a `expecting 2 plurals, found 1` warning.

Reply:
Added to my todo list.

Reply:
Okay, It looks like that Pontoon supports plural forms only for `.po` files which make this test invalid, so I removed it.